src/test/java/io/confluent/connect/jdbc/source/integration/MSSQLDateTimeIT.java

```diff
@@ -0,0 +1,314 @@
+     * This test is for a datetime to datetime2 comparison error in MSSQL Server (2016 and after).
+     * Datetime is converted recursively to a higer precision (eg- 3.33 MS would be 3.33333 ),
+     * while java.sql.Timestamp converts by tacking on 0s (3.33000).
+     */
+
+    @Test
+    public void verifyDATETIME2DoesNotKillTask() throws Exception {
+        props = configProperties(true);
+        // And verify that datetime in a non-timestamp column does not kill task.
+        createDateTimeTable("DATETIME2", "DATETIME", true);
+        ZoneId utc = ZoneId.of("UTC");
+        // Create topic in Kafka
+        KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+        // Configure source connector
+        configureAndWaitForConnector();
+
+        java.sql.Timestamp t = java.sql.Timestamp.from(
+                ZonedDateTime.of(2016, 12, 8, 19, 34, 56, 2000000, utc).toInstant()
+        );
+
+        writeSingleRowWithTimestamp(t, false, true);
+
+        Thread.sleep(Duration.ofSeconds(30).toMillis());
+        for (String topic: KAFKA_TOPICS) {
+            ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+                    NUM_RECORDS_PRODUCED,
+                    CONSUME_MAX_DURATION_MS,
+                    topic);
+            //Assert that records in topic == NUM_RECORDS_PRODUCED
+            assertEquals(NUM_RECORDS_PRODUCED, records.count());
+
+        }
+        // clean up table
+        deleteTable();
+    }
+
+    private Map<String, String> configProperties(boolean timestampModeOnly) {
+        // Create a hashmap to setup source connector config properties
+        Map<String, String> props = new HashMap<>();
+        props.put(CONNECTOR_CLASS_CONFIG, "JdbcSourceConnector");
+        props.put(CONNECTION_URL_CONFIG, MSSQL_URL);
+        props.put(CONNECTION_USER_CONFIG, "sa");
+        props.put(CONNECTION_PASSWORD_CONFIG, "reallyStrongPwd123");
+        props.put(TABLE_WHITELIST_CONFIG, MSSQL_Table);
+
+        if (timestampModeOnly) {
+            props.put(MODE_CONFIG, MODE_TIMESTAMP);
+        } else {
+            props.put(MODE_CONFIG, MODE_TIMESTAMP_INCREMENTING);
+            props.put(INCREMENTING_COLUMN_NAME_CONFIG, "ict");
+        }
+
+        props.put(TIMESTAMP_COLUMN_NAME_CONFIG, "start_time");
+        props.put(TOPIC_PREFIX_CONFIG, "test-");
+        props.put(POLL_INTERVAL_MS_CONFIG, "30");
+        props.put(TASKS_MAX_CONFIG, Integer.toString(TASKS_MAX));
+        props.put(KEY_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+        props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+        props.put(CONFLUENT_TOPIC_BOOTSTRAP_SERVERS_CONFIG, connect.kafka().bootstrapServers());
+        props.put(CONFLUENT_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+
+        return props;
+    }
+
+    private void createDateTimeTable(
+            String timestampColType,
+            String regularColType,
+            boolean timestampModeOnly
```

Comment:
I would probably just pass in the CREATE string as an arg instead of this bool, it's kind of hard to read